src/lib/operators/table_scan/column_is_null_table_scan_impl.cpp

```diff
@@ -151,6 +166,78 @@ bool ColumnIsNullTableScanImpl::_matches_none(const BaseValueSegment& segment) c
   }
 }
 
+template <typename BaseSegmentType>
```

Comment:
Now that the interface is the same: do we need the templated function? Why not accept an abstract segment?

Reply:
Unfortunately, the interface is not the same.